lib/addons/p5.dom.js

```diff
@@ -308,6 +308,87 @@
   };
 
   /**
+   * Creates a checkbox &lt;input&gt;&lt;/input&gt; element in the DOM.
+   *
+   * @method createCheckbox
+   * @param  {String} [label] label displayed after checkbox
+   * @param  {boolean} [value] value of the checkbox; checked is true, unchecked is false
+   * @return {Object/p5.Element} pointer to p5.Element holding created
+   *                           node
+   */
+  p5.prototype.createCheckbox = function() {
+    var elt = document.createElement('input');
+    elt.type = 'checkbox';
+    //checkbox must be wrapped in p5.Element before label so that label appears after
+    var self = addElement(elt, this);
+    self.checked = function(){
+      if (arguments.length === 0){
+        return elt.checked;
+      }else if(arguments[0] === true){
+        elt.checked = true;
+      }else{
+        elt.checked = false;
+      }
+      return self;
+    };
+    if (arguments.length > 0){
+      var label = document.createElement('label');
+      elt.setAttribute('id', '_id');
+      label.htmlFor = '_id';
```

Comment:
aha got it. would we run into problems if they called the createCheckbox method more than once, because we'd end up with multiple elements with same id?